<launch>
	<arg name="use_stabilizer" default="true" />

<!--	<include file="$(find seabee3_machines)/client-server.machine" />-->

<!--	<include file="$(find seabee3_launch)/seabee3_dead_reckoning.launch" /> -->

	<include if="$(arg use_stabilizer)" file="$(find seabee3_stabilizer)/launch/seabee3_stabilizer.launch" />

	<node pkg="seabee3_driver" type="seabee3_driver" name="seabee3_driver" output="screen" >
		<remap if="$(arg use_stabilizer)" from="~motor_cntl" to="/seabee3/motor_cntl_raw" />
		<remap unless="$(arg use_stabilizer)" from="~motor_cntl" to="/seabee3/motor_cntl" />

		<param name="rate" type="double" value="30" />
		<param name="use_pid_assist" type="bool" value="true" />
		
		<param name="cmd_vel_timeout" type="double" value="0.5" />
		
		<param name="global_frame" type="string" value="/landmark_map" />

		<param name="cmd_vel_conversion" type="int" value="1" />
		
		<param name="motor_val_deadzone" type="double" value="5.0" />
		<param name="motor_val_min" type="double" value="15.0" />
		<param name="motor_val_max" type="double" value="100.0" />
		
		<!-- ========== PID tuning ========== -->
		
		<param name="depth_err_cap" type="int" value="100" />
		<param name="roll_err_cap" type="double" value="25.0" />
		<param name="pitch_err_cap" type="double" value="25.0" />
		<param name="heading_err_cap" type="double" value="50.0" />
		
		<param name="pid/linear/x/p" type="double" value="100.0" />
		<param name="pid/linear/x/i" type="double" value="0.01" />
		<param name="pid/linear/x/d" type="double" value="0.6" />
		
		<param name="pid/linear/y/p" type="double" value="100.0" />
		<param name="pid/linear/y/i" type="double" value="0.01" />
		<param name="pid/linear/y/d" type="double" value="0.6" />
		
		<param name="pid/linear/z/p" type="double" value="50.0" />
		<param name="pid/linear/z/i" type="double" value="0.001" />
		<param name="pid/linear/z/d" type="double" value="0.2" />
		
		<param name="pid/angular/x/p" type="double" value="2.5" />
		<param name="pid/angular/x/i" type="double" value="0.05" />
		<param name="pid/angular/x/d" type="double" value="0.2" />
		
		<param name="pid/angular/y/p" type="double" value="2.5" />
		<param name="pid/angular/y/i" type="double" value="0.05" />
		<param name="pid/angular/y/d" type="double" value="0.2" />
		
		<param name="pid/angular/z/p" type="double" value="114.59" />
		<param name="pid/angular/z/i" type="double" value="0.05" />
		<param name="pid/angular/z/d" type="double" value="0.6" />
		
		<param name="pid_i_max" type="double" value="1.0" />
		<param name="pid_i_min" type="double" value="-1.0" />
		
		<!-- ========== Motor/Behavior Tuning ========== -->
		
		<param name="speed_m1_dir" type="double" value="1.0" />
		<param name="speed_m2_dir" type="double" value="1.0" />
		
		<param name="strafe_m1_dir" type="double" value="-1.0" />
		<param name="strafe_m2_dir" type="double" value="1.0" />
		
		<param name="depth_m1_dir" type="double" value="-1.0" />
		<param name="depth_m2_dir" type="double" value="-1.0" />
		
		<param name="roll_m1_dir" type="double" value="-1.0" />
		<param name="roll_m2_dir" type="double" value="1.0" />
		
		<param name="yaw_m1_dir" type="double" value="-1.0" />
		<param name="yaw_m2_dir" type="double" value="-1.0" />
		
		<param name="speed_axis_dir" type="double" value="-1.0" />
		<param name="strafe_axis_dir" type="double" value="1.0" />
		<param name="depth_axis_dir" type="double" value="1.0" />
		<param name="roll_axis_dir" type="double" value="-1.0" />
		<param name="pitch_axis_dir" type="double" value="-1.0" />
		<param name="heading_axis_dir" type="double" value="-1.0" />
	</node>
</launch>
